### PR TITLE
Extend EC2 joining method to other AWS regions

### PR DESCRIPTION
**Pull request by tigrato**
This PR extends the support for EC2 joining method for the following AWS regions:

- ap-south-2
- ap-southeast-3
- ap-southeast-4
- ca-west-1
- eu-central-2
- eu-south-2
- il-central-1
- me-central-1

Changelog: Expand the EC2 joining process to include newly created AWS regions

### CODE REVIEW

**Review by reedloden — COMMENTED** [review]
Perhaps sort this alphabetically to make it easier to see if a region is missing?

**Comment by tigrato** [comment]
> Perhaps sort this alphabetically to make it easier to see if a region is missing?

done in [d65cc04](https://github.com/gravitational/teleport/pull/39045/commits/d65cc048fb385cbaa004c37c0acf43f6f1dbcfd5)

File: lib/auth/aws_certs.go [hunk 1]
```diff
@@ -560,5 +560,118 @@ MNvXDi/ErzgrHGSpcvmGHiOhMf3UzChMWbIr6udoDlMbSIO7+8F+jUJkh4Xl1lKb
 YeN5fsLZp7T/6YvbFSPpmbn1YoE2vKtuGKxObRrhU3h4JHdp1Zel1pZ6lh5iM0ec
 SD11SximGIYCjfZpRqI3q50mbxCd7ckULz+UUPwLrfOds4VrVVSj+x0ZdY19Plv2
 9shw5ez6Cn7E3IfzqNHO
+-----END CERTIFICATE-----`),
+	"ap-south-2": []byte(`-----BEGIN CERTIFICATE-----
```
Comment (reedloden) [inline lib/auth/aws_certs.go]: These are the RSA certificates, not the RSA 2048 ones.

https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/regions-certs.html

There are three tabs for every region -- DSA, RSA, and RSA 2048. We need to use the RSA 2048 ones.

**Review by reedloden — CHANGES_REQUESTED** [review]

File: lib/auth/aws_certs.go (hunk 1 above)
Comment (tigrato) [reply lib/auth/aws_certs.go]: My bad. Fixed in [f2ea30a](https://github.com/gravitational/teleport/pull/39045/commits/f2ea30a90618a1609927e56b626e75c5553be49c)

**Review by reedloden — APPROVED** [review]
Spot checked each one against https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/regions-certs.html (RSA 2048 tab) and confirmed they look correct. Thanks!

**Review by mdwn — APPROVED** [review]

**Review by strideynet — APPROVED** [review]

**Pull request merged**